File: src/dotnet/src/Product.Api/ProductApi.Core/ProductPriceBracket.cs
// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
// This product includes software developed at Datadog (https://www.datadoghq.com/).
// Copyright 2024 Datadog, Inc.

namespace ProductApi.Core;

public class ProductPriceBracket(decimal quantity, decimal price)
{
    public decimal Price { get; } = price;
    public decimal Quantity { get; } = quantity;
}

File: src/java/product-api-container/src/main/java/com/product/api/container/core/ProductPriceBracket.java
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2024 Datadog, Inc.
 */

package com.product.api.container.core;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class ProductPriceBracket {
    private final Double quantity;
    private final Double price;
    
    public ProductPriceBracket(){
        this.price = 0.0;
        this.quantity = -1.0;
    }
    
    public ProductPriceBracket(Double quantity, Double price) {
        this.quantity = quantity;
        this.price = price;
    }

    public Double getPrice() {
        return price;
    }

    public Double getQuantity() {
        return quantity;
    }
}
